src/Inventor/Qt/devices/SoQtMouse.cpp

```diff
@@ -117,9 +117,9 @@ SoQtMouse::translateEvent(QEvent * event)
 
 #ifdef HAVE_SOMOUSEBUTTONEVENT_BUTTON5
   if (wheelevent) {
-    if (wheelevent->delta() > 0)
+    if (wheelevent->angleDelta().y() > 0)
```

Comment:
This need to check the Qt version as it was introduced in Qt 5 and SoQt is usable with Qt 4 too, e.g.

#if QT_VERSION >= 0x050C00
    if (wheelevent->angleDelta().y() > 0)
      PRIVATE(this)->buttonevent->setButton(wheelevent->inverted() ? SoMouseButtonEvent::BUTTON5 : SoMouseButtonEvent::BUTTON4);
    else if (wheelevent->angleDelta().y() < 0)
      PRIVATE(this)->buttonevent->setButton(wheelevent->inverted() ? SoMouseButtonEvent::BUTTON4 : SoMouseButtonEvent::BUTTON5);
#elif QT_VERSION >= 0x050000
    if (wheelevent->angleDelta().y() > 0)
      PRIVATE(this)->buttonevent->setButton(SoMouseButtonEvent::BUTTON4);
    else if (wheelevent->angleDelta().y() < 0)
      PRIVATE(this)->buttonevent->setButton(SoMouseButtonEvent::BUTTON5);
#else 
    if (wheelevent->delta() > 0)
      PRIVATE(this)->buttonevent->setButton(SoMouseButtonEvent::BUTTON4);
    else if (wheelevent->delta() < 0)
      PRIVATE(this)->buttonevent->setButton(SoMouseButtonEvent::BUTTON5);
#endif

Reply:
Regarding the first case: Shouldn't that be 0x050700 instead of 0x050c00, as QWheelEvent::inverted() was introduced in 5.7?

Reply:
You are right. My bad. I was looking at the wrong place for isInverted documentation